pkg/shell/machines/machines.js

```diff
@@ -73,6 +73,9 @@ export function get_init_superuser_for_options(options) {
 function Machines() {
     const self = this;
 
+    // HACK - tell TypeScript that addEventListener exists.
+    self.addEventListener = null;
```

Comment:
... so that this hack isn't necessary. Isn't the whole point of typing to encode this properly, rather than this runtime duck typing stuff?

Reply:
Yup, fine for me. Turning this all into properly declared classes/interfaces is certainly a follow-up, I was just curious about the unknown thing above.

Reply:
I have now added proper types for machines/machines.